Tests/TestingApps/PaywallsTester/PaywallsTester/Data/ApplicationData.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  ApplicationData.swift
+//
+//
+//  Created by Nacho Soto on 12/11/23.
+//
+
+import Foundation
+
+import OSLog
+
+@Observable
+final class ApplicationData {
+
+    private(set) var authentication: Authentication = .unknown {
+        didSet {
+            Self.logger.info("Changed authentication: \(String(describing: self.authentication))")
+        }
+    }
+
+    @MainActor
+    func loadApplicationData() async throws {
+        self.authentication = .unknown
+
+        do {
+            self.authentication = .signedIn(try await self.manager.loadApplicationData())
+        } catch ApplicationManager.Error.unauthenticated {
+            Self.logger.warning("Received unauthentication error when loading application data")
+            self.authentication = .signedOut
+        }
+    }
+
+    func signOut() {
+        self.manager.signOut()
+        self.authentication = .signedOut
+    }
+
+    @ObservationIgnored
+    private var manager: ApplicationManagerType = ApplicationManager()
+    private static let logger = Logger(subsystem: Bundle.main.bundleIdentifier ?? "com.revenuecat.PaywallsTester",
```

Comment:
bundleId will never be nil, right? Wondering whether keeping around the hardcoded version is overkill

Reply:
In terms of our app it'll never be nil, but it is an optional property, so we could force-unwrap instead. Thoughts?

Reply:
I mean, I would usually not force-unwrap unless it's totally safe, but... it's totally safe

Reply:
Done!